lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -785,6 +791,19 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif query_string:
+        try:
+            vtype = ipaddr(query, 'type')
+            if vtype == 'address':
+                v = ipaddr(query, 'cidr')
+            elif vtype == 'network':
+                v = ipaddr(query, 'subnet')
+            query = netaddr.IPNetwork(v)
+        except NETADDR_ERRORS as err:
+            raise errors.AnsibleFilterError(err)
```

Comment:
This is minor, but I like to keep exception chain when re-raising new ones. This is how you do it in forward-compatible manner:
```python
from ansible.module_utils import six
...
	except NETADDR_ERRORS as err:
            six.raise_from(errors.AnsibleFilterError(err), err)
```